packages/server/lib/controllers/files.js

```diff
@@ -85,17 +84,15 @@ module.exports = {
       if (spec === '__all') {
         debug('returning all specs')
 
-        return specsUtil.find(config)
+        return specsUtil.default.findSpecs(config)
         .then(R.tap((specs) => {
           return debug('found __all specs %o', specs)
         }))
         .filter(specFilterFn)
         .filter((foundSpec) => {
-          if (componentTestingEnabled) {
-            return foundSpec.specType === specTypeFilter
-          }
-
-          return true
+          return componentTestingEnabled
```

Comment:
This is the "main" change. Instead of assuming e2e, then doing some CT specific mods IF in CT mode, we do not assume e2e or CT. Instead, we assume you can either be in e2e or CT mode, and are explicit about what spec(s) to get.